pom.xml

```diff
@@ -5,7 +5,7 @@
 
     <groupId>io.projectdiscovery</groupId>
     <artifactId>nuclei-burp-plugin</artifactId>
-    <version>1.1.3-SNAPSHOT</version>
+    <version>1.1.3-SNAPSHOT</version> <!-- update version before release -->
```

Comment:
The release can be done with Maven, which automatically updates the version.

#### Prepare a release:
```shell
mvn release:clean release:prepare
```
1. Set the version number you want to release in <_MAJOR.MINOR.PATCH_> format (e.g. `1.0.0`)
2. Set the SCM release tag: (e.g. `1.0.0`)
3. Set the new development version (e.g. `1.0.1-SNAPSHOT`)

Check the created commits and tag to make sure everything looks as expected:
   ```shell
   git log
   git show HEAD
   git show HEAD^
   git tag -l # list tags
   ```

#### Rollback the local release:
```shell
mvn release:rollback # or "git reset HEAD^^ --hard"
git tag -d <tag_name_to_delete>
```

Perform the release:
```shell
mvn release:perform 
```

#### Push the new release to GitHub:
```shell
git push origin master
git push origin <tag_name>
```

Of course all of the above could be made fully automated, but I am not sure that's such a good idea.

Instead of hard-coding the versions for the artifact uploading you could extract the current version from maven using the following snippet:

```yaml
- name: Get artifact version
      id: get-version
      run: echo "::set-output name=version::$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)"
```

Furthermore the reason why I've used upload artifact and on push was because it was convenient for other maintainers to test the plugin, without having to manually build it after every commit.